src/pymor/algorithms/eigs.py

```diff
@@ -0,0 +1,308 @@
+# This file is part of the pyMOR project (http://www.pymor.org).
+# Copyright 2013-2020 pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+
+import numpy as np
+import scipy.linalg as spla
+
+from pymor.algorithms.gram_schmidt import gram_schmidt
+from pymor.core.logger import getLogger
+from pymor.operators.constructions import IdentityOperator
+from pymor.operators.interface import Operator
+
+
+def eigs(A, E=None, k=3, which='LM', b=None, l=None, maxiter=1000, tol=1e-13, seed=0):
+    """Approximate a few eigenvalues of an |Operator|.
+
+    Computes `k` eigenvalues `w` with corresponding eigenvectors `v` which solve
+    the eigenvalue problem
+
+    .. math::
+        A v_i = w_i v_i
+
+    or the generalized eigenvalue problem
+
+    .. math::
+        A v_i = w_i E v_i
+
+    if `E` is not `None`.
+
+    The implementation is based on Algorithm 4.2 in [RL95]_.
+
+    Parameters
+    ----------
+    A
+        The real |Operator| for which the eigenvalues are to be computed.
```

Comment:
In the current implementation yes, because it is assumed that complex eigenvalues always appear in pairs of complex conjugates. I assume in general it is not possible to check if A is complex and then run a slightly different algorithm, or is it?

Reply:
At the moment, unfortunately not.